mrmustard/physics/ansatze.py

```diff
@@ -380,3 +380,114 @@ def __and__(self, other: PolyExpAnsatz) -> PolyExpAnsatz:
         bs = [math.concat([b1, b2], axis=-1) for b1 in self.b for b2 in other.b]
         cs = [math.outer(c1, c2) for c1 in self.c for c2 in other.c]
         return self.__class__(As, bs, cs)
+
+class ArrayAnsatz(Ansatz):
+    r"""Represents the ansatz as a multidimensional array.
+    """
+    def __init__(self, array: Batch[Tensor]):
+        r"""Note that the array is batched."""
+        self.array = array
+    
+    @property
+    def array(self) -> Batch[Tensor]:
+        r"""
+        The array.
+        """
+        return self.array
```

Comment:
If you save `self.array` in the init, you do not need the property. You need the property (and potentially the setter) if you store `self._array`. So pick one :)

Reply:
Got it ! Thanks!